programs/stake/src/stake_state.rs

```diff
@@ -757,33 +761,71 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
+                        split.data_len()? as u64,
+                    );
 
                     // verify enough lamports for rent in new stake with the split
                     if split.lamports()? + lamports < meta.rent_exempt_reserve
-                     // verify enough lamports left in previous stake and not full withdrawal
+                    // verify enough lamports left in previous stake and not full withdrawal
                         || (lamports + meta.rent_exempt_reserve > self.lamports()? && lamports != self.lamports()?)
+                    // verify that the same rent_exempt_reserve is sufficient to make the new
+                    // stake account rent exempt needed because the new uninitialized may be any
+                    // size larger than std::mem::size_of::<StakeState>()
+                        || split_rent_exempt_reserve > meta.rent_exempt_reserve
                     {
                         return Err(InstructionError::InsufficientFunds);
                     }
                     // split the stake, subtract rent_exempt_balance unless
-                    //  the destination account already has those lamports
-                    //  in place.
-                    // this could represent a small loss of staked lamports
-                    //  if the split account starts out with a zero balance
-                    let split_stake = stake.split(
-                        lamports - meta.rent_exempt_reserve.saturating_sub(split.lamports()?),
-                    )?;
+                    // the destination account already has those lamports
+                    // in place.
+                    // this means that the new stake account will have a stake equivalent to
+                    // lamports minus rent_exempt_reserve if it starts out with a zero balance
+                    let (remaining_stake_delta, split_stake_amount) = if lamports
+                        == self.lamports()?
+                    {
+                        // If split amount equals the full source stake, the new split stake must
+                        // equal the same amount, regardless of any current lamport balance in the
+                        // split account. Since split accounts retain the state of their source
+                        // account, this prevents any magic activation of stake by prefunding the
+                        // split account.
+                        (
+                            lamports - meta.rent_exempt_reserve,
+                            lamports - meta.rent_exempt_reserve,
+                        )
+                    } else {
+                        // Otherwise, the new split stake should reflect the entire split
+                        // requested, less any lamports needed to cover the rent-exempt reserve
+                        (
+                            lamports,
+                            lamports - meta.rent_exempt_reserve.saturating_sub(split.lamports()?),
```

Comment:
I think this must be `lamports - split_rent_exempt_reserve.saturating_sub(split.lamports()?)`, right?

Reply:
Ah, yes, correct.